lib/path/default.nix

```diff
@@ -100,6 +102,20 @@ let
     # An empty string is not a valid relative path, so we need to return a `.` when we have no components
     (if components == [] then "." else concatStringsSep "/" components);
 
+  # Deconstruct a path value type into:
+  # - root: The filesystem root of the path, generally `/`
+  # - components: All the path components of the path
+  #
+  # This is similar to `splitString "/" (toString path)` but safer
+  # because it can distinguish different filesystem roots
+  deconstructPath =
+    let
+      go = components: path:
```

Comment:
The naming nit from the other PR still holds 😛

Reply:
I'll change it to `recurse` [as in the other PR](https://github.com/NixOS/nixpkgs/pull/209375#discussion_r1069220818) :)